File: strategy/c-sharp/PaypalPayment.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Strategy
{
    // UMl diyagramındaki ConcreteStrategy sınıfına denk gelmektedir.
    // UML diyagramındaki Strategy arayüzüne denk gelen arayüzü uyguluyor.
    class PaypalPayment : IPayment
    {
        private string _email;
        private string _password;

        public PaypalPayment(string email, string password)
        {
            _email = email;
            _password = password;
        }

        public void Pay(int amount)
        {
            // Kredi kartı ödeme işlemlerinin gerçekleştiği yer.
            Console.WriteLine("Paid by Paypal.");
        }
    }
}

File: strategy/go/paypal-payment.go
package main

import "fmt"

// UMl diyagramındaki ConcreteStrategy sınıfına denk gelmektedir.
// UML diyagramındaki Strategy arayüzüne denk gelen arayüzü uyguluyor.
type paypalPayment struct {
	email    string
	password string
}

func NewPayPalPayment(email, password string) Payment {
	return paypalPayment{
		email:    email,
		password: password,
	}
}

func (p paypalPayment) Pay(amount float64) {
	// Paypal ödeme işlemlerinin gerçekleştiği yer.
	fmt.Printf("Paypal payment: %.2f\n", amount)
}
